src/plugins/size/size.c

```diff
@@ -0,0 +1,140 @@
+/**
+ * @file
+ *
+ * @brief Source for size plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include "size.h"
+
+#include <kdbhelper.h>
+#include <stdio.h>
+#include <string.h>
+
+
+#define DEFAULT_SEPARATOR "x"
+
+static int transform (KeySet * appendKS, const Key * key, const char * sep)
+{
+	char * value = elektraStrDup (keyString (key));
+	if (!value) return 0;
+	char * dptr = strstr (value, sep);
+	if (!dptr) return 0;
```

Comment:
Yes, it would only be a validation error if on writing out size or height is missing (and no default exists). When parsing, emitting a warning would be needed though, we should not silently ignore metadata.